hacking/target-mm32.md

```diff
@@ -0,0 +1,102 @@
+
+## HOWTO Add new MM32 Cortex-M0 or Cortex-M0+
+
+To add support for a new MM32 Cortex-M0 or Cortex-M0+ processor, first check the product id at address 0x40013400.
+E.g. for MM32SPIN05:
+```
+(gdb) mon swd
+Target voltage: 
+Available Targets:
+No. Att Driver
+*** 1   Unknown ARM Cortex-M Designer 0x43b Part ID 0x471 M0
+(gdb) at 1
+Attaching to Remote target
+warning: No executable has been specified and target does not support
+determining executable automatically.  Try using the "file" command.
+warning: while parsing target memory map (at line 1): Required element <memory> is missing
+0xfffffffe in ?? ()
+(gdb) set mem inaccessible-by-default off
+(gdb) x 0x40013400
+0x40013400:	0xcc4460b1
+```
+The product id can also be found by searching the - English or Chinese - user manual for "DEV_ID". DEV_ID is the 32-bit Cortex JEDEC-106 ID code. 
+
+With the product id known, add a new case to the switch statement in file src/target/stm32f1.c, function mm32l0xx_probe() :
+```
+case 0xcc4460b1U:
+        name = "MM32SPIN05";
+        flash_kbyte = 32;
+        ram_kbyte = 4;
+        break;
+```
+where flash_kbyte is the amount of flash at 0x08000000, and ram_kbyte is the amount of ram at 0x20000000.
+
+## HOWTO Add new MM32 Cortex-M3 or STAR-MC1
+
+To add support for a new MM32 Cortex-M3 or STAR-MC1 processor, first check the product id at address 0x40007080.
+E.g. for MM32F3273:
+```
+(gdb) mon swd
+(gdb) at 1
+(gdb) set mem inaccessible-by-default off
+(gdb) x 0x40007080
+0x40007080:	0xcc9aa0e7
+```
+With the product id known, add a new case to the switch statement in file src/target/stm32f1.c, function mm32f3xx_probe() :
+```
+case 0xcc9aa0e7U:
+        name = "MM32F3270";
+        flash_kbyte = 512;
+        ram1_kbyte = 128;
+        break;
+```
+where flash_kbyte is the amount of flash at 0x08000000, ram1_kbyte is the amount of ram at 0x20000000. For STAR-MC1 processors with ram at 0x30000000, set ram2_kbyte to the amount of ram at 0x30000000.
+
+If different models exist that have the same product id but different amounts of flash or ram, choose the model with the largest amount of flash or ram.
+
+Lastly, open a github "Pull Request" to add the processor to the black magic git.
+
+## Resources
+
+- [HAL library and sample code](https://www.mindmotion.com.cn/en/development_tools/) contains start-up code for Keil and IAR
+- [mm32_startup](https://github.com/iclite/mm32_startup/) start-up code for mm32 with gcc
+- [libgcc.a patch](https://github.com/koendv/MM32SPIN27-Arduino/tree/main/mm32/mm32-libgcc) to use hardware division with gcc on MM32 Cortex-M0. 
+- Arduino for [MM32SPIN27](https://github.com/koendv/MM32SPIN27-Arduino)
+- Arduino for [MM32F031](http://47.106.166.129/fenyi/4_channel_SPDT_relay_wirmware/)
+- SeekFree [libraries](https://gitee.com/seekfree/projects)  for MM32F527X, MM32F327X, MM32SPIN27, MM32SPIN360 and others
+
+_not truncated_
```

Comment:
This looks weird on the website render and isn't something any other similar pages do, please drop this line.